### PR TITLE
Small cleanups, insignificant terrain get speedup

### PR DESCRIPTION
**Pull request by fewizz**

### CODE REVIEW

File: src/main/java/twilightforest/TwilightForestMod.java [hunk 1]
```diff
@@ -88,16 +88,16 @@ public void postInit(FMLPostInitializationEvent evt) {
...
 			TwilightForestMod.LOGGER.warn("Detected that the configured dimension id '{}' is being used.  Using backup ID.  It is recommended that you configure this mod to use a unique dimension ID.", TFConfig.dimension.dimensionID);
-			DimensionManager.registerDimension(TwilightForestMod.backupdimensionID, TwilightForestMod.dimType);
-			TFConfig.dimension.dimensionID = TwilightForestMod.backupdimensionID;
+			DimensionManager.registerDimension(TwilightForestMod.BACKUP_DIM_ID, TwilightForestMod.dimType);
+			TFConfig.dimension.dimensionID = TwilightForestMod.BACKUP_DIM_ID;
 		}
 
-		if (Loader.isModLoaded("Thaumcraft")) {
-			//FIXME: Reenable this once Thaumcraft is available.
+		//FIXME: Reenable this once Thaumcraft is available.
```
Comment (liach) [inline src/main/java/twilightforest/TwilightForestMod.java]: It's not going to be back

File: src/main/java/twilightforest/entity/passive/EntityTFBunny.java [hunk 2]
```diff
@@ -41,6 +42,8 @@ protected void initEntityAI() {
 		this.tasks.addTask(0, new EntityAISwimming(this));
 		this.tasks.addTask(1, new EntityAIPanic(this, 2.0F));
 		this.tasks.addTask(2, new EntityAITempt(this, 1.0F, Items.WHEAT_SEEDS, true));
+		this.tasks.addTask(3, new EntityAITempt(this, 1.0D, Items.CARROT, true));
+        this.tasks.addTask(3, new EntityAITempt(this, 1.0D, Items.GOLDEN_CARROT, true));
```
Comment (liach) [inline src/main/java/twilightforest/entity/passive/EntityTFBunny.java]: spaces vs tabs

**Review by liach — APPROVED** [review]

File: src/main/java/twilightforest/TFFeature.java [hunk 3]
```diff
@@ -281,7 +281,7 @@ public TFFeature addWaterCreature(Class<? extends EntityLiving> monsterClass, in
 	public static TFFeature getFeatureDirectlyAt(int chunkX, int chunkZ, World world) {
 
 		if (world != null && world.getBiomeProvider() instanceof TFBiomeProvider) {
-			TFBiomeProvider tfManager = (TFBiomeProvider) world.getBiomeProvider();
+			//TFBiomeProvider tfManager = (TFBiomeProvider) world.getBiomeProvider();
```
Comment (williewillus) [inline src/main/java/twilightforest/TFFeature.java]: ? if the variable is unused just remove the line

File: src/main/java/twilightforest/block/enums/BossVariant.java [hunk 4]
```diff
@@ -19,6 +19,9 @@
 	MINOSHROOM(false, TileEntityTFMinoshroomSpawner.class),
 	ALPHA_YETI(false, TileEntityTFAlphaYetiSpawner.class);
 
+	// Caching, cos vm each time creates new array when calling values()
```
Comment (williewillus) [inline src/main/java/twilightforest/block/enums/BossVariant.java]: comment unnecessary

File: src/main/java/twilightforest/client/renderer/TFSkyRenderer.java [hunk 5]
```diff
@@ -34,52 +35,52 @@ public TFSkyRenderer() {
 	@SideOnly(Side.CLIENT)
```
Comment (williewillus) [inline src/main/java/twilightforest/client/renderer/TFSkyRenderer.java]: do not change this method or anything with [VanillaCopy] on it, the code needs to diff cleanly (or near-cleanly) with vanilla so we can track changes in vanilla properly

File: src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java [hunk 6]
```diff
@@ -224,34 +230,38 @@ public void generateTerrain2(int chunkX, int chunkZ, ChunkPrimer primer) {
...
-			for (int az = 0; az < 5; ++az) {
+			int biomeForGenIndex = ax + 2 + 2 * 10;
+			
+			for (int az = 0; az < 5; ++az, biomeForGenIndex += 10) {
 				float totalVariation = 0.0F;
 				float totalHeight = 0.0F;
 				float totalFactor = 0.0F;
 				byte two = 2;
-				Biome biomegenbase = this.biomesForGeneration[ax + 2 + (az + 2) * 10];
+				Biome biomegenbase = this.biomesForGeneration[biomeForGenIndex/*ax + 2 + (az + 2) * 10*/];
```
Comment (williewillus) [inline src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java]: don't leave random things commented in, just remove it if you're confident in the change

File: src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java [hunk 7]
```diff
@@ -224,34 +230,38 @@ public void generateTerrain2(int chunkX, int chunkZ, ChunkPrimer primer) {
...
 						float heightVariation = biomegenbase1.getHeightVariation();
 
-						if (this.field_147435_p == WorldType.AMPLIFIED && rootHeight > 0.0F) {
+						if (this.worldType == WorldType.AMPLIFIED && rootHeight > 0.0F) {
 							rootHeight = 1.0F + rootHeight * 2.0F;
 							heightVariation = 1.0F + heightVariation * 4.0F;
 						}
 
-						float heightFactor = this.parabolicField[ox + 2 + (oz + 2) * 5] / (rootHeight + 2.0F);
+						float heightFactor = this.parabolicField[pfIndex/*ox + 2 + (oz + 2) * 5*/] / (rootHeight + 2.0F);
```
Comment (williewillus) [inline src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java]: same thing here, remove comment

File: src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java [hunk 8]
```diff
@@ -675,16 +685,16 @@ private void deformTerrainForTrollCloud2(ChunkPrimer primer, TFFeature nearFeatu
 		}
 	}
 
-	private float pseudoRand(int bx, int bz) {
+	/*private float pseudoRand(int bx, int bz) {
 		Random rand = new Random(this.world.getSeed() + (bx * 321534781) ^ (bz * 756839));
 		rand.setSeed(rand.nextLong());
 		return rand.nextFloat();
-	}
+	}*/
```
Comment (williewillus) [inline src/main/java/twilightforest/world/ChunkGeneratorTwilightForest.java]: if this method is unused just remove it

**Review by williewillus — CHANGES_REQUESTED** [review]

File: src/main/java/twilightforest/client/renderer/TFSkyRenderer.java [hunk 9]
```diff
@@ -20,6 +20,8 @@
```
Comment (williewillus) [inline src/main/java/twilightforest/client/renderer/TFSkyRenderer.java:20]: can you undo all changes to this file?

File: src/main/java/twilightforest/TwilightForestMod.java [hunk 10]
```diff
@@ -88,15 +88,8 @@ public void postInit(FMLPostInitializationEvent evt) {
 			DimensionManager.registerDimension(TFConfig.dimension.dimensionID, TwilightForestMod.dimType);
 		} else {
 			TwilightForestMod.LOGGER.warn("Detected that the configured dimension id '{}' is being used.  Using backup ID.  It is recommended that you configure this mod to use a unique dimension ID.", TFConfig.dimension.dimensionID);
-			DimensionManager.registerDimension(TwilightForestMod.backupdimensionID, TwilightForestMod.dimType);
-			TFConfig.dimension.dimensionID = TwilightForestMod.backupdimensionID;
-		}
-
-		if (Loader.isModLoaded("Thaumcraft")) {
-			//FIXME: Reenable this once Thaumcraft is available.
-			//registerThaumcraftIntegration();
```
Comment (williewillus) [inline src/main/java/twilightforest/TwilightForestMod.java:97]: If you're going to remove this remove the actual `registerThaumcraftIntegration` method as well

File: src/main/java/twilightforest/entity/passive/EntityTFBunny.java [hunk 11]
```diff
@@ -41,6 +41,8 @@ protected void initEntityAI() {
 		this.tasks.addTask(0, new EntityAISwimming(this));
 		this.tasks.addTask(1, new EntityAIPanic(this, 2.0F));
 		this.tasks.addTask(2, new EntityAITempt(this, 1.0F, Items.WHEAT_SEEDS, true));
+		this.tasks.addTask(3, new EntityAITempt(this, 1.0D, Items.CARROT, true));
+		this.tasks.addTask(3, new EntityAITempt(this, 1.0D, Items.GOLDEN_CARROT, true));
```
Comment (williewillus) [inline src/main/java/twilightforest/entity/passive/EntityTFBunny.java:45]: add dandelions and remove wheat seeds (see vanilla EntityRabbit)

**Review by williewillus — CHANGES_REQUESTED** [review]

**Pull request closed without merging**